### ISSUE TITLE
Feature request: reset or limit Miniview size

### ISSUE THREAD

**Issue by ioxua-os** [iamlemec/miniview#11]
So I had this problem where sometimes the ctrl modifier gets stuck (still trying to figure the problem out) and every mouse movement turns into a window resize.
Some moments later the window turned too small to even be resized again, so a reset size or size limit would be useful.
I'm willing to implement the feature myself, just need to know what is preferred

**Comment by iamlemec** [comment]
Thanks for the report! Sorry for the slow reply here.

Yeah, I've bumped into that issue before. I'm able to trigger it if I move the mouse outside of the window really fast. Actually, it seems like the best way to handle it is to end the resize if there's a leave-event from the mouse.

Agreed, it's probably good to have a size reset option. One idea is to turn the "reset opacity" menu item into a general reset button. It's probably not used too often and this way we keep menu clutter to a minimum.

**Comment by ioxua-os** [comment]
@iamlemec no problem!

I can think turning the "reset opacity" button into a more general reset button is a great idea! 

I think I can implement that, if you are willing to accept a pull request. It's my first time messing with gnome so I need to study a little but I think I can handle it

**Comment by iamlemec** [comment]
That sounds great! Let me know if you need any pointers along the way. I'm happy to help.

In addition to a general reset option, I was thinking having a minimum absolute size of say 10 in either dimension might be good too.

**Comment by iamlemec** [comment]
@kctang yeah, that's quite the pickle you're in there. It seems like unloading and loading the extension doesn't reset the size/position/opacity ephemera. You'll need to reset gnome-shell to really reset these parameters (until we get the general reset option). You can do that with Alt+F2, then run the pseudo-command "r". That only works in X11 though, in Wayland you'll need to log out and log back in.

As a backstop until I'm able to release a new version, the latest commit here on Github makes it much less likely that you'll end up in this situation by getting stuck in resize mode.

**Comment by kctang** [comment]
thanks for the feedback @iamlemec. no worries... i'll check back on miniview when the reset feature is in. ;-)

**Comment by iamlemec** [comment]
Ok, new version is here and up on extensions.gnome.org! It has the new reset feature and also a change that should stop this issue from happening in the first place.

**Comment by kctang** [comment]
just gave it a quick test...works well, although i noticed that when i resize the miniview, i need to do it slowly or some of the resizing movements are not registered. v.minor issue..doesn't bother me much.

thanks!
